jina/drivers/encode.py

```diff
@@ -34,3 +34,22 @@ def _apply_all(self, docs: 'DocumentSet', *args, **kwargs) -> None:
                 raise LengthMismatchException(msg)
             for doc, embedding in zip(docs_pts, embeds):
                 doc.embedding = embedding
+
+
+class ScipySparseEncodeDriver(FlatRecursiveMixin, BaseEncodeDriver):
```

Comment:
is there a unified way to work with this with other sparse vectors? Isn't it better to detect the type in `EncodeDriver`? There may be no need for an extra `encodedriver`?

Reply:
We could do this in EncodeDriver adding a generator that allow us to iterate over the rows of different 'tensor types' with the same unified interface:

```

            for doc, embedding in zip(docs_pts, embeds):
                doc.embedding = embedding
```
by
```
            for doc, embedding in zip(docs_pts, matrix_iterator(embeds)):
                doc.embedding = embedding
```

where 

```
def matrix_iterator(X):
    n_rows = X.shape[0]
    for i in range(n_rows):
        if scipy.sparse.issparse(X):
            yield X.getrow(i)
        else:
            yield X[i]
```

Reply:
Something like this, but we do not even need to changr the iterator, we can do the if else one level above (just be careful on how to check dependencies)

Reply:
have a private `assign_embeddings` method that works differently for numpy arrays than for sparse arrays